### PR TITLE
TFLite: Missing activation in built-in Mul operation

### PR DESCRIPTION
**Pull request by guy-david**
ReLU activation went missing despite appearing in the graph as an activation function in the Mul operation.
Hope this fixes a TODO marked by @ekaterinaignasheva

### CODE REVIEW

File: tensorflow/lite/delegates/gpu/common/model_builder.cc [hunk 1]
```diff
@@ -1392,6 +1393,8 @@ class MulOperationParser : public TFLiteOperationParser {
     const bool runtime_tensor0 = !constant_tensor0;
     const bool runtime_tensor1 = !constant_tensor1;
 
+    Node* node = graph->NewNode();
+
```
Comment (ekaterinaignasheva) [inline tensorflow/lite/delegates/gpu/common/model_builder.cc]: Please, create a new node only when operation is created. If function returns with InternalError the graph will be left corrupted.

File: tensorflow/lite/delegates/gpu/common/model_builder.cc [hunk 2]
```diff
@@ -1406,27 +1409,36 @@ class MulOperationParser : public TFLiteOperationParser {
         input_tensor0 = 1;
         input_tensor1 = 0;
       }
-      return ParseApplyMask(input_tensor0, input_tensor1, graph, reader);
+      RETURN_IF_ERROR(ParseApplyMask(input_tensor0, input_tensor1, node, graph, reader));
+    }
+    else
+    {
```
Comment (ekaterinaignasheva) [inline tensorflow/lite/delegates/gpu/common/model_builder.cc]: Please, follow the google code style:
} else {

File: tensorflow/lite/delegates/gpu/common/model_builder.cc [hunk 3]
```diff
@@ -1406,27 +1409,36 @@ class MulOperationParser : public TFLiteOperationParser {
...
-    int constant_tensor = 1;
-    TfLiteIntArray* constant_dims = input1->dims;
-    if (constant_tensor0 && runtime_tensor1) {
-      runtime_tensor = 1;
-      constant_tensor = 0;
-      constant_dims = input0->dims;
+    const auto* tf_options =
+        reinterpret_cast<const TfLiteMulParams*>(tflite_node->builtin_data);
+    if (!tf_options) {
+      return InternalError("Missing tflite params");
```
Comment (ekaterinaignasheva) [inline tensorflow/lite/delegates/gpu/common/model_builder.cc]: Please, use more descriptive error messages, like "Missing TfLiteMulParams"

**Review by ekaterinaignasheva-zz — CHANGES_REQUESTED** [review]
Please, fix the comments.

File: tensorflow/lite/delegates/gpu/common/model_builder.cc (hunk 1 above)
Comment (guydavid) [reply tensorflow/lite/delegates/gpu/common/model_builder.cc]: I moved the params check a bit higher but it still seems that corruption is unavoidable.

File: tensorflow/lite/delegates/gpu/common/model_builder.cc (hunk 2 above)
Comment (guydavid) [reply tensorflow/lite/delegates/gpu/common/model_builder.cc]: Thanks

File: tensorflow/lite/delegates/gpu/common/model_builder.cc (hunk 3 above)
Comment (guydavid) [reply tensorflow/lite/delegates/gpu/common/model_builder.cc]: I went on and corrected other messages

File: tensorflow/lite/delegates/gpu/common/model_builder.cc (hunk 2 above)
Comment (impjdi) [reply tensorflow/lite/delegates/gpu/common/model_builder.cc]: I'm still seeing

}
else
{

**Review by impjdi — DISMISSED** [review]

**Comment by gbaned** [comment]
@guydavid Can you please resolve conflicts? Thanks!

File: tensorflow/lite/delegates/gpu/common/model_builder.cc [hunk 4]
```diff
@@ -1412,21 +1413,32 @@ class MulOperationParser : public TFLiteOperationParser {
         input_tensor0 = 1;
         input_tensor1 = 0;
       }
-      return ParseApplyMask(node, input_tensor0, input_tensor1, graph, reader);
+      RETURN_IF_ERROR(ParseApplyMask(node, input_tensor0, input_tensor1, graph,
+                                     reader));
+    }
```
Comment (impjdi) [inline tensorflow/lite/delegates/gpu/common/model_builder.cc]: can you merge L.1418 & L.1419?  Otherwise, everything looks fine.

Comment (guydavid) [reply tensorflow/lite/delegates/gpu/common/model_builder.cc]: Thanks

**Review by impjdi — APPROVED** [review]

**Review by rthadur — APPROVED** [review]

**Comment by rthadur** [comment]
Seems auto-merge is not happening but the changes are now committed so we can close this. Thank you for the PR.
 @chanshah thank you for importing this PR internally.

**Pull request merged**